doc/developer_guide/developer_guide.md

```diff
@@ -0,0 +1,121 @@
+# Developer Guide
+
+This document contains developer information on how to build, run, modify and publish this Lua project.
+
+## Prerequisites
+
+This project needs a Lua interpreter &ge; Lua 5.1.
+
+To install the dependencies you also need the [LuaRocks](https://luarocks.org/) package manager.
+
+## Sources and tests
+
+You find the implementation under `src` and the tests under `test`.
+
+## Versioning
+
+We use semantic versioning, meaning that the major version indicates backward compatibility, the minor version behavior changes and the fix version everything else.
+
+The version is defined in the LuaRock specification file in the root directory.
+
+```
+<module-name>-<version>-<package-revision>.rockspec
+```
+
+Example:
+
+```
+mymodule-15.8.0-1.rockspec
+```
+
+As you can see, the version is part of the filename already. Unfortunately you also have to change it _inside_ the file.
+
+```
+-- ...
+version = "15.8.0-1"
+
+source = {
+    url = -- project URL on GitHub
+    tag = "15.8.0"
+}
+```
+
+Additionally, you need to change the version number in the module itself under `src/<module-path>`:
+
+```lua
+-- ...
+
+local M = {
+    VERSION = "15.8.0",
+    -- ...
+}
+```
+
+Finally, each new version needs an entry in the changelog under `doc/changes/changes_<version>.md` and a link to that file from `doc/changes/changelog.md`.
+
+## Running the tests
+
+To run all the tests in the project use:
+
+```bash
+tools/runtests.sh
+```
+
+This script also runs the static code analysis. Note that this is the same script used in the [continuous integration](#continuous-integration) builds.
+
+The unit tests are written using [LuaLUnit](https://github.com/bluebird75/luaunit). The are therefore directly executable with Lua:
+
+```bash
+lua tests/<test-name>.lua
+```
+
+## Building the Package
+
+Refer to the [LuaRocks documentation](https://github.com/luarocks/luarocks/wiki/Documentation) for detailed information about package management in Lua.
+
+To build and install the package locally use the command below. "Locally" in this context means as a regular user for yourself as opposed to globally as administrator.
+
+```bash
+luarocks make --local
+```
+
+If you have administrator privileges on your machine and prefer a global installation, simply omit the `--local` switch in all commands.
+
+**Warning:** Be careful, when you have packages install that you also work on, you have to make sure that the work-in-progress code is in the front of the package path. Otherwise you end up debugging an installed package instead of your code!
```

Comment:
```suggestion
**Warning:** Be careful, when you have packages installed that you also work on, you have to make sure that the work-in-progress code is in the front of the package path. Otherwise you end up debugging an installed package instead of your code!
```